salt/modules/yumpkg.py

```diff
@@ -1159,7 +1159,19 @@ def _add_common_args(cmd):
 
     errors = []
 
+    hold_pkgs = list_holds(full=False)
+    log.error(hold_pkgs)
+    to_hold = []
     if targets:
+        if len(targets) == 1:
+            to_hold.append(pkgname)
```

Comment:
This will result in the package being held if there was only one targeted package. This doesn't look right. Also, why are you using ``pkgname`` instead of ``targets[0]``?